.github/workflows/kubeapps-main-reusable.yaml

```diff
@@ -0,0 +1,754 @@
+# Copyright 2022 the Kubeapps contributors
+# SPDX-License-Identifier: Apache-2.0
+
+on:
+  workflow_call:
+    inputs:
+      run_gke_tests:
+        type: boolean
+        required: false
+        default: false
+
+env:
+  CHARTMUSEUM_VERSION: "3.9.1"
+  CHARTS_REPO_ORIGINAL: "bitnami/charts"
+  BRANCH_CHARTS_REPO_ORIGINAL: "main"
+  CHARTS_REPO_FORKED: "kubeapps-bot/charts"
+  BRANCH_CHARTS_REPO_FORKED: "main"
+  CI_BOT_USERNAME: "kubeapps-bot"
+  CI_BOT_EMAIL: "tanzu-kubeapps-team@vmware.com"
+  CI_BOT_GPG: "80B6EB16B1328FB18DFF2A073EBA68F3347E319D"
+  # DEBUG_MODE allows to activate some SSH debugging steps, and modify the verbosity level of some scripts (eg. e2e-tests.sh)
+  DEBUG_MODE: "true"
+  # FIXME: DEV_MODE is used to enable dev mode, used to activate several hacks along the code:
+  # - In sync_chart_from/to_bitnami jobs to file shadow PRs while we maintain
+  # - To add a suffix to the tag of the generated docker images, to prevent collision and overwriting with those generated from CircleCI
+  # - To add a suffix to the name of the generated GKE clusters to prevent collision with those generated and used from CircleCI
+  # - Probably some other things I don't remember right now :)
+  #
+  # CircleCI and GHA pipelines are running in parallel. This should be removed once the CircleCI pipeline is decommissioned.
+  DEV_MODE: true
+  SSH_KEY_KUBEAPPS_DEPLOY_FILENAME: "id_rsa_kubeapps_deploy_key"
+  SSH_KEY_FORKED_CHARTS_DEPLOY_FILENAME: "id_rsa_forked_charts_deploy_key"
+  KUBEAPPS_REPO: "vmware-tanzu/kubeapps"
+  BRANCH_KUBEAPPS_REPO: "main"
+  README_GENERATOR_REPO: "bitnami-labs/readme-generator-for-helm"
+  DOCKER_VERSION: "20.10.18"
+  DOCKER_REGISTRY_VERSION: "2.8.1"
+  GOLANG_VERSION: "1.19.2"
+  HELM_VERSION_MIN: "v3.2.0"
+  HELM_VERSION_STABLE: "v3.10.1"
+  GITHUB_VERSION: "2.18.1"
+  IMAGES_TO_PUSH: "apprepository-controller dashboard asset-syncer pinniped-proxy kubeapps-apis"
+  # IMG_DEV_TAG is the tags used for the Kubeapps docker images. Ideally there should be an IMG_PROD_TAG
+  # but its value is dynamic and GitHub actions doesn't support it in the `env` block, so it is generated
+  # as an output of the `setup` job.
+  IMG_DEV_TAG: "build-${{ github.sha }}"
+  # Apart from using a dev tag we use a different image ID to avoid polluting the tag history of the production tag
+  IMG_MODIFIER: "-ci-gha"
+  IMG_PREFIX: "kubeapps/"
+  # We use IMG_PREFIX_FOR_FORKS for development purposes, it's used when the workflow is run from a fork of the kubeapps repo
+  IMG_PREFIX_FOR_FORKS: "beni0888/"
```

Comment:
Leftover? Or is it still useful while working on the GHA?

Reply:
Well, it is useful anytime when someone needs to develop the pipeline and push docker images to an alternative Docker Hub workspace. When the pipeline is triggered inside a fork, it will use the value from this variable. At this moment it contains my personal workspace, but it can be changed to whatever is needed.